lib/logflare/backends/adaptor/clickhouse_adaptor/query_templates.ex

```diff
@@ -0,0 +1,152 @@
+defmodule Logflare.Backends.Adaptor.ClickhouseAdaptor.QueryTemplates do
+  @moduledoc """
+  Common query templates utilized by the `ClickhouseAdaptor`.
+  """
+
+  import Logflare.Guards
+
+  @key_type_counts_view_name "mv_key_type_counts_per_minute"
+  @key_type_counts_table_name "key_type_counts_per_minute"
+  @default_ttl_days 90
+
+  @doc """
+  Generates a ClickHouse query statement to provision an ingest table for logs.
+
+  ###Options
+
+  - `:database` - (Optional) Will produce a fully qualified `<database>.<table>` string when provided with a value. Defaults to `nil`.
+  - `:ttl_days` - (Optional) Will add a TTL statement to the table creation query. Defaults to `90`. `nil` will disable the TTL.
+
+  """
+  @spec create_log_ingest_table_statement(table :: String.t(), opts :: Keyword.t()) :: String.t()
+  def create_log_ingest_table_statement(table, opts \\ [])
+      when is_non_empty_binary(table) and is_list(opts) do
+    database = Keyword.get(opts, :database, nil)
+    ttl_days_temp = Keyword.get(opts, :ttl_days, @default_ttl_days)
+
+    ttl_days =
+      if is_pos_integer(ttl_days_temp) do
+        ttl_days_temp
+      else
+        nil
+      end
+
+    db_table_string =
+      if is_non_empty_binary(database) do
+        "#{database}.#{table}"
+      else
+        "#{table}"
+      end
+
+    Enum.join([
+      """
+      CREATE TABLE IF NOT EXISTS #{db_table_string} (
```

Comment:
needs some sanitization checks if we're interpolating it directly into the query. maybe just a simple check to ensure no DDL keywords.

Reply:
I have now removed that old `table` attribute from the adaptor to minimize confusion.

Reply:
But at the end of the day, if you really want DDL keywords to be filtered - just tell me.

Reply:
ah made the comment as the database value comes from the user but doesn't seem to have any validation. 
i tihnk its fine to ignore then

Reply:
👋 

ClickHouse supports "identifier" params for tables names and such.

```elixir
iex> Ch.query(pool, "create table {table:Identifier} (name String) engine MergeTree order by ()", %{"table" => "; drop table Students"})

iex> Ch.query!(pool, "select name from system.tables order by metadata_modification_time desc limit 1").rows
#==> [["; drop table Students"]]
```

I was just passing through :)